### PR TITLE
Polishing the final version

### PR DESCRIPTION
**Pull request by jacobrs**
comments + abstraction of complicated digit stack logic

### CODE REVIEW

File: app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/Calculator.kt [hunk 1]
```diff
@@ -4,19 +4,22 @@ import java.util.Stack
 
 class Calculator {
 
-    val formula: Stack<FormulaItem> = Stack()
+    private val formula: Stack<FormulaItem> = Stack()
 
-    fun pushOperand(operand: Double): Unit {
```
Comment (fvalela1) [inline app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/Calculator.kt:9]: Glad to see kotlinisms being implemented (y)!

File: app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/Calculator.kt [hunk 2]
```diff
@@ -45,22 +48,24 @@ class Calculator {
...
             formula.pop()
     }
 
-    fun clear(): Unit {
+    fun clear() {
         formula.clear()
     }
 
+    // Objects are similar to singletons or static classes. This companion object shares the same
+    // name as its primary user, the Calculator class.
```
Comment (fvalela1) [inline app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/Calculator.kt:68]: Good explanation, especially for people not coming to the tutorial/review

File: app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/MainActivity.kt [hunk 3]
```diff
@@ -3,147 +3,54 @@ package com.example.francescovalela.androidtutorialwinter2018
...
-import java.util.*
 
 class MainActivity : AppCompatActivity() {
 
-    private val calc = Calculator()
-    private val digitStack: Stack<Double> = Stack()
-    private var lastOperator: Char? = null
-    private var lastOperand: Double? = null
+    // This adapter handles the digit stack so we can abstract it in this activity
+    private val userInterfaceAdapter: UserInterfaceAdapter = UserInterfaceAdapter(this)
```
Comment (fvalela1) [inline app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/MainActivity.kt:11]: Nice!!!

File: app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/MainActivity.kt [hunk 4]
```diff
@@ -3,147 +3,54 @@ package com.example.francescovalela.androidtutorialwinter2018
...
-
-    private fun genericDigitListener(digit: Double){
-        if(lastOperator == null) {
-            calc.clear()
-            lastOperand = null
-        }
-        digitStack.push(digit)
-        refreshFormulaView()
+        btn_clear.setOnClickListener { userInterfaceAdapter.clear() }
+        btn_equals.setOnClickListener{ userInterfaceAdapter.equals() }
```
Comment (fvalela1) [inline app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/MainActivity.kt:29]: We may still have to explain the ‘setOnClickListener’ portion. Should we consider having one “regular” way of doing it?

**Review by fvalela1 — APPROVED** [review]

File: app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/MainActivity.kt (hunk 4 above)
Comment (jacobrs) [reply app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/MainActivity.kt:29]: not sure what you mean by one "regular" way of doing it.

Comment (fvalela1) [reply app/src/main/java/com/example/francescovalela/androidtutorialwinter2018/MainActivity.kt:29]: as in not calling another class to do it, but now that I think about it, it's fine. I think we can show on the powerpoint the very basic basic correlation between activity_main.xml and mainactivity.kt

**Pull request merged**